Add scripts to setup CC with qemu on a single node

amshinde opened the pull request:
Scripts create a VM image with TDX support, installs required dependencies such as docker and kubernetes, sets up proxy setttings for systemd services and provides instructions to launch a kubernetes cluster.

Note:
This is a work-in-progress PR intended to get initial feedback on the workflow. While parts of it are verified, there still needs to be additional verification done.

amshinde committed:
Add scripts to setup CC with qemu on a single node

Scripts create a VM image with TDX support, install required
dependencies and start qemu with confidential cluster running on it.

Signed-off-by: Archana Shinde <archana.m.shinde@intel.com>

amshinde commented:
@ruomengh Please take a look.

chavafg commented on deployment/bare-metal/README.md:
```diff
@@ -0,0 +1,74 @@
+This guide introduces how to setup an Intel TDX host on Ubuntu 24.04 and a TD VM with
+a single node kubernetes cluster running on it.
+Follow these instructions to setup Intel TDX host, create a TD image, boot the TD and run a 
+kubernestes cluster within the TD.
```
typo: kubernestes

chavafg commented on deployment/bare-metal/README.md:
```diff
@@ -0,0 +1,74 @@
+
+### Setup host
+
+We first need to install ageneric Ubuntu 24.04 server image, install necessay packages to turn
```
typos: ageneric and necessay

chavafg commented on deployment/bare-metal/create_k8s_node.sh:
```diff
@@ -0,0 +1,54 @@
+#!/bin/bash
+#
+# Copyright (c) 2020 Intel Corporation
```
should this be 2024? same for all scripts.

ruomengh commented on deployment/bare-metal/setup_cc.sh:
```diff
@@ -0,0 +1,170 @@
+}
+
+install_pre_reqs() {
+	sudo -E apt install -y qemu-utils guestfs-tools virtinst genisoimage libvirt-daemon-system libvirt-daemon cloud-init
```
@amshinde Qemu and libvirt should already be installed via Canonical before running setup_cc.sh.

amshinde replied:
Yes, I just added this in case these are uninstalled. In any case if these are already installed, this will just be no-op.

ruomengh commented on deployment/bare-metal/setup_cc.sh:
```diff
@@ -0,0 +1,170 @@
+	mv "${tdx_repo}/guest-tools/image/${img_name}" /tmp/${GUEST_IMG}
+}
+
+create_tdx_image_cc_image_writer() {
```
This function seems not in use anywhere.

ruomengh commented on deployment/bare-metal/README.md:
````diff
@@ -0,0 +1,74 @@
+
+Once you have logged in the TD VM, run the following script to start a single node kubernetes cluster:
+```
+$ /home/tdx/launch_cc.sh
````
Where is the file "launch_cc.sh"?

amshinde replied:
This should be `create_k8s_node.sh`. Have changed it.

ruomengh commented:
@amshinde Seems the launch_k8s or launch_k3s part is not implemented yet. Also a few inline comments. Thanks.

chavafg commented on deployment/bare-metal/install_dependencies.sh:
```diff
@@ -0,0 +1,180 @@
+#!/bin/bash
```
missing license header?

chavafg commented on deployment/bare-metal/install_dependencies.sh:
```diff
@@ -0,0 +1,180 @@
+	apt-get update > /dev/null
+
+	# install docker
+	apt-get install -y docker-ce docker-ce-cli containerd.io docker-buildx-plugin docker-compose-plugin
```
should this be sudo? 
same for other apt-get commands below

also, I see a combination of `apt` and `apt-get` in the script. should we use just one?

amshinde replied:
I have changed all invocations of apt-get to apt. Also got rid of sudo by checking for effective user id at the beginning of the script.

chavafg commented on deployment/bare-metal/install_dependencies.sh:
```diff
@@ -0,0 +1,180 @@
+	
+	install_docker
+	install_helm
+	#install_pip
```
uncomment?

amshinde commented:
@ruomengh I have addressed your review comments last week itself. Any pending typos including changing all usage of `apt-get` to `apt` has been done as well. Please take a look.

ruomengh reviewed (approved):
LGTM
